sdk/samples/iot/aziot_realtek_amebaD/aziot_realtek_amebaD.ino

```diff
@@ -0,0 +1,385 @@
+  }
+  
+  *out_decoded_bytes = az_span_create(az_span_ptr(decoded_bytes), (int32_t)len);
+}
+
+static void iot_sample_generate_sas_base64_encoded_signed_signature(
+    az_span sas_base64_encoded_key,
+    az_span sas_signature,
+    az_span sas_base64_encoded_signed_signature,
+    az_span* out_sas_base64_encoded_signed_signature)
+{
+  //IOT_SAMPLE_PRECONDITION_NOT_NULL(out_sas_base64_encoded_signed_signature);
+
+  // Decode the sas base64 encoded key to use for HMAC signing.
+  char sas_decoded_key_buffer[32];
+  az_span sas_decoded_key = AZ_SPAN_FROM_BUFFER(sas_decoded_key_buffer);
+  decode_base64_bytes(sas_base64_encoded_key, sas_decoded_key, &sas_decoded_key);
+
+  // HMAC-SHA256 sign the signature with the decoded key.
+  char sas_hmac256_signed_signature_buffer[32];
+  az_span sas_hmac256_signed_signature = AZ_SPAN_FROM_BUFFER(sas_hmac256_signed_signature_buffer);
+  hmac_sha256_sign_signature(sas_decoded_key, sas_signature, sas_hmac256_signed_signature, &sas_hmac256_signed_signature);
+
+  // Base64 encode the result of the HMAC signing.
+  base64_encode_bytes(
+    sas_hmac256_signed_signature,
+    sas_base64_encoded_signed_signature,
+    out_sas_base64_encoded_signed_signature);
+}
+
+static void generate_sas_key(void)
+{
+
+  az_result rc;
+  // Create the POSIX expiration time from input minutes.
+  uint64_t sas_duration = iot_sample_get_epoch_expiration_time_from_minutes(SAS_TOKEN_EXPIRY_IN_MINUTES);
+
+  char b64enc_hmacsha256_signature[64];
+
+  // Get the signature that will later be signed with the decoded key.
+  az_span sas_signature = AZ_SPAN_FROM_BUFFER(signature);
+  rc = az_iot_hub_client_sas_get_signature(
+    &hub_client, sas_duration, sas_signature, &sas_signature);
+  if (az_result_failed(rc))
+  {
+    Serial.print("Could not get the signature for SAS key: az_result return code ");
+    Serial.println(rc);
+  }
+
+  // Generate the encoded, signed signature (b64 encoded, HMAC-SHA256 signing).
+  az_span sas_base64_encoded_signed_signature = AZ_SPAN_FROM_BUFFER(b64enc_hmacsha256_signature);
+  iot_sample_generate_sas_base64_encoded_signed_signature(
+    //AZ_SPAN_FROM_BUFFER(device_key),
+    device_key,
+    sas_signature,
+    sas_base64_encoded_signed_signature,
+    &sas_base64_encoded_signed_signature);
+  
+  // Get the resulting MQTT password, passing the base64 encoded, HMAC signed bytes.
+  size_t mqtt_password_length;
+  rc = az_iot_hub_client_sas_get_password(
+    &hub_client,
+    sas_duration,
+    sas_base64_encoded_signed_signature,
+    AZ_SPAN_EMPTY,
+    sas_token,
+    sizeof(sas_token),
+    &sas_token_length);
+  if (az_result_failed(rc))
+  {
+    Serial.print("Could not get the password: az_result return code ");
+    Serial.println(rc);
+  }
+}
+
+static int connectToAzureIoTHub()
+{
+  size_t client_id_length;
+  char mqtt_client_id[128];
+  if (az_result_failed(az_iot_hub_client_get_client_id(
+          &hub_client, mqtt_client_id, sizeof(mqtt_client_id) - 1, &client_id_length)))
```

Comment:
?
```suggestion
          &hub_client, mqtt_client_id, sizeofarray(mqtt_client_id) - 1, &client_id_length)))
```

Reply:
Not needed because sizeof(mqtt_client_id) == sizeof(char) == 1.